File: Go/Slice/remove-duplicates-from-slice.go
package main

import (
	"fmt"
)

func main() {
	// Create a slice of strings
	randomSlice := []string{"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J"}
	// Remove all the duplicates from the slice
	fmt.Println(removeDuplicatesFromSlice(randomSlice))
}

// Remove all the duplicates from a slice and return the slice.
func removeDuplicatesFromSlice(slice []string) []string {
	check := make(map[string]bool)
	var newReturnSlice []string
	for _, content := range slice {
		if !check[content] { {
			check[content] = true
			newReturnSlice = append(newReturnSlice, content)
		}
	}
	return newReturnSlice
}

File: Dart/Slice/remove_duplicates_from_slice.dart
import "dart:core";

void main() {
  // Create a new list of strings.
  final List<String> randomSlice = <String>["Bob","Bob", "Alice", "Joe"];
  // Remove all duplicate elements from the list.
  print(removeDuplicatesFromSlice(randomSlice));
}

// Remove all the duplicates element from a slice.
List<dynamic> removeDuplicatesFromSlice(final List<dynamic> slice) {
  final List<dynamic> uniqueElements = <dynamic>[];
  for (final dynamic element in slice) {
    if (!uniqueElements.contains(element)) {
      uniqueElements.add(element);
    }
  }
  return uniqueElements;
}
